### DISCUSSION TITLE
For + object = RangeError: Invalid array length

### DISCUSSION THREAD

**Discussion by milahu** [solidjs/solid#721] [Ideas]
when i pass an object to `<For>`

```jsx
<For each={someObject}>{(val, idx) => val}</For>
```

then solidjs throws the confusing error message ```RangeError: Invalid array length```
which ideally should be something like `TypeError: expected array`

**Comment by ryansolid** [comment]
I see because it uses length to determine all behavior and an undefined sets it down a path where it tries to initialize a new array with the length undefined. Hmm.. I can probably add one more check that will no op it. But that's probably worse than no error. I kinda prefer not to isArray check it everytime unnecessarily. I suppose I could add that check in dev as a happy medium.
